main.py

```
@@ -1,32 +1,92 @@
+        return self._memgraph_port
+
+    @property
+    def pull_frequency_seconds(self) -> int:
+        return self._pull_frequency_seconds
+
+
+config_singleton = None
 
-# Create a metric to track time spent and requests made.
-REQUEST_TIME = Summary('request_processing_seconds', 'Time spent processing request')
-SIZE_CTR = Counter('size_counter', 'Size counter description')
-SIZE_GG = Gauge('size_gauge', 'Size gauge description')
-HISTOGRAM = Histogram('histie', 'Histogram description')
 
-@REQUEST_TIME.time()
 def pull_metrics():
-    """ Pull Memgraph metrics """
-    res = requests.get("http://localhost:9092/")
+    # We first want to fetch metrics from the Memgraph's HTTP endpoint
+    res = requests.get(
+        f"{config_singleton.memgraph_endpoint_url}:{config_singleton.memgraph_port}"
+    )
+
+    print(f"Pulled metrics: status code {res.status_code}")
     if res.status_code != 200:
-        raise Exception("Status code is not 200!")
-    
+        raise Exception(
+            f"Status code is not 200, but {res.status_code}, please check running services!"
+        )
+
     json_data = res.json()
-    size = json_data["size"]
-    SIZE_CTR.inc(size)
-    SIZE_GG.set(size)
-    HISTOGRAM.observe(random.random())
 
+    update_metrics(json_data)
+
+
+if __name__ == "__main__":
+    # Parse the configuration for starting the service and retrieve data from correct endpoints
+    config_singleton = Config.from_yaml_file()
+    start_http_server(config_singleton.exporter_port)
 
-if __name__ == '__main__':
-    # Start up the server to expose the metrics.
-    start_http_server(8000)
-    # Generate some requests.
+    # Continuously fetch metrics
     while True:
-        time.sleep(5)
-        pull_metrics()
+        time.sleep(config_singleton.pull_frequency_seconds)
+        try:
+            pull_metrics()
+        except Exception as e:
+            print("Exception occurred while pulling Memgraph metrics!")
+            print(e)
```

I think we should introduce `logging` module for this part

logging.basicConfig(format='%(asctime)-15s [%(levelname)s]: %(message)s')
logger = logging.getLogger('prometheus_handler')
logger.setLevel(logging.INFO)

And do we want to close `main.py` on all errors or are there specific ones to which we want to raise an exception

---

I actually now set it to continue, not exit. I think it's not good to exit when I think of it, but continue pinging until memgraph is back